### PR TITLE
Add termux-am-2

### PR DESCRIPTION
**Pull request by tareksander**
Client to use the new am implementation from this PR: https://github.com/termux/termux-app/pull/2458

### CODE REVIEW

**Review by thunder-coding — CHANGES_REQUESTED** [review]
https://github.com/tareksander/termux-am-2/blob/20f10c7396a9a3951067b3786c5e11637ce53b80/CMakeLists.txt#L12: It would make sense to use TERMUX_BASE_DIR instead of prefix. This will remove the need of "../" from https://github.com/tareksander/termux-am-2/blob/20f10c7396a9a3951067b3786c5e11637ce53b80/termux-am.c#L15

File: packages/termux-am-2/build.sh
Comment (Grimler91) [inline packages/termux-am-2/build.sh:2]: Please remove the empty lines

File: packages/termux-am-2/build.sh
Comment (Grimler91) [inline packages/termux-am-2/build.sh:1]: Maybe termux-am-socket, or similar, could be a more descriptive name for the repo and package?

**Review by Grimler91 — CHANGES_REQUESTED** [review]

File: packages/termux-am-2/build.sh
Comment (thunder-coding) [reply packages/termux-am-2/build.sh:1]: I feel, the repo can be transferred to `@termux`. It will make maintaince of the utility easier for us.

**Comment by Grimler91** [comment]
Please put the socket in $PREFIX/var/run instead of /data/data/com.termux/files, that's what we tend to make other packages use for sockets

**Comment by Grimler91** [comment]
What do you think about moving the repo to the github.com/termux namespace and continue development there? We would then send an organisation invite so that you get write access to everything. 

We can then also move your [termux-gui plugin](https://github.com/tareksander/termux-gui) to github.com/termux, making it an "official" plugin (unless you prefer to keep it separate)

**Comment by tareksander** [comment]
When transferring ownership, do I have to use "Termux", "termux" or "@termux" as the organization name?

**Comment by Grimler91** [comment]
> When transferring ownership, do I have to use "Termux", "termux" or "@termux" as the organization name?

If it would be in github.com/termux then it would under the termux organization on github, thereby grouping it with the other apps and termux-related repositories. All copyright would still belong to you, and maintainer/author fields can still point to you. 
In any case, I understand if you want to keep termux-gui separate (just let us know if you at some point want to move it to this organization)

Are you fine with us moving termux-am-2 to github.com/termux/?

**Comment by tareksander** [comment]
I'll transfer termux-am-2 now, then you can also give the repo a more appropriate name, I couldn't come up with a better one.

**Comment by tareksander** [comment]
> Are you fine with us moving termux-am-2 to github.com/termux/?

It says "cannot find new owner github.com/termux".

**Comment by agnostic-apollo** [comment]
> It says "cannot find new owner github.com/termux".

Use `termux`.

**Comment by tareksander** [comment]
> Use termux.

Now it says "You don’t have the permission to create public repositories on termux"

**Comment by agnostic-apollo** [comment]
> Please put the socket in $PREFIX/var/run instead of /data/data/com.termux/files,

Issue with that is that if `$PREFIX` doesn't exist and `TermuxApplication` just creates the `var/run` directory, then it would be incorrectly assumed that `$PREFIX` exists and bootstrap installation wouldn't be done. That may even result in some backward compatibility issues. API commands wouldn't run since their scripts like `termux-reset` and `am` wrapper wouldn't exist either. Maybe socket and `var/run` shouldn't be created if `$PREFIX` doesn't exist, that may work. Need to think more on this.

**Comment by Grimler91** [comment]
> > Use termux.
> 
> Now it says "You don’t have the permission to create public repositories on termux"

I cloned and pushed it here: https://github.com/termux/termux-am-socket

Also sent you an invite so that we can give you access to the repo, should be in your mail

**Comment by Grimler91** [comment]
> > Please put the socket in $PREFIX/var/run instead of /data/data/com.termux/files,
> 
> Issue with that is that if `$PREFIX` doesn't exist and `TermuxApplication` just creates the `var/run` directory, then it would be incorrectly assumed that `$PREFIX` exists and bootstrap installation wouldn't be done. 

Hm, good point. Maybe /data/data/com.termux/files, or /data/data/com.termux/files/sockets or similar is better then

**Comment by agnostic-apollo** [comment]
Yeah, `/data/data/com.termux/files/sockets` would be better.

**Comment by agnostic-apollo** [comment]
Maybe some other generic name could be used instead of `sockets` if the directory is to be used for other stuff in future, like `api` or something even more generic.

**Comment by tareksander** [comment]
> I cloned and pushed it here:

While you're at it you can also do that for the am library repo I made that is used as a module in my termux-app PR:  https://github.com/tareksander/termux-am-library

**Comment by agnostic-apollo** [comment]
You should be able to create the repo yourself once you accept the invitation. The `termux-app` pull request would need to be reviewed before it can be merged. It likely has conflicts with my local commits. I'll try to look into it soon, busy with lot of stuff.

**Comment by tareksander** [comment]
> You should be able to create the repo yourself once you accept the invitation.

It still gives me the same error and when trying to select termux when creating a new repo it says "insufficient permission".

**Comment by agnostic-apollo** [comment]
You have  been added as a member, not owner, hence the issue. Grimler, can you handle the repo, I am away from laptop atm.

**Comment by Grimler91** [comment]
> > You should be able to create the repo yourself once you accept the invitation.
> 
> It still gives me the same error and when trying to select termux when creating a new repo it says "insufficient permission".

Fixed, please try again!

**Comment by tareksander** [comment]
> Fixed, please try again!

Thanks, it works now.

**Pull request merged**
